apps/client/src/features/operator/edit-modal/EditModal.tsx

```diff
@@ -16,33 +16,45 @@ export default function EditModal(props: EditModalProps) {
 
   const { updateCustomField } = useEventAction();
   const [loading, setLoading] = useState(false);
-  const inputRef = useRef<HTMLTextAreaElement | null>(null);
+  const inputRef = useRef<HTMLTextAreaElement[]>(new Array<HTMLTextAreaElement>());
 
   const handleSave = async () => {
     setLoading(true);
-    const newValue = inputRef.current?.value;
-    if (newValue === undefined) {
-      return;
-    }
+    inputRef.current?.forEach(async (element) => {
+      if (element.dataset.field && element.defaultValue != element.value) {
+        await updateCustomField(event.id, element.dataset.field, element.value);
```

Comment:
I fear this may be a bit too complicated. some tips:

- check if inputRef.current is null and short circuit, like
```ts
  if (!inputRef.current) return;
```

- we likely dont want to make several requests to update all the promises. could we generate a single patch object and patch at once?